src/pushsource/_impl/backend/pub_source/pub_source.py

```diff
@@ -25,6 +25,7 @@ def __init__(
         task_id,
         threads=4,
         timeout=60 * 60 * 4,
+        filename="images.json",
```

Comment:
I'm fundamentally disagreeing with this change.

Prior to the change I would describe PubSource as:

- It gives you all the push items found on a particular Pub task.
- The current implementation _happens to_ only give you AMI push items, but that's not something built-in to the API.  It's just that the other push item types which could potentially be supported have not yet been implemented.  They might become supported later, with no API changes needed.

Now after this change it seems to be rather:

- It gives you all the push items _relating to one specified attachment_ on a particular Pub task.

I don't think that is a good API.  It's not going to work for other types of push items which ideally would be supported and don't appear in attached JSON files like AMI do.  It's inconsistent with other sources (e.g. errata source loads all push items on the advisory - it doesn't require the caller to specify some subset of items to be loaded).  The worst is that it forces the caller to know about Pub task implementation details (caller has to know which filename makes sense for their case) rather than keeping it hidden inside this library.

Could you explain the bigger picture of what you want to accomplish?  I read the JIRA issue but it wasn't clear.

If the issue is that some tasks have (only) images.json while others have (only) clouds.json, then I suggest PubSource should try to load both, and use the one which exists (or yield no items if neither exists).  That would be more consistent with the way the other sources work.

Reply:
Thanks for the feedback. I'll close this and reopen one that is more in line with this.